cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -61,8 +71,17 @@ public DatasetInstanceHandler(DatasetInstanceService instanceService) {
   @Path("/data/datasets/")
   public void list(HttpRequest request, HttpResponder responder,
                    @PathParam("namespace-id") String namespaceId) throws Exception {
+
     responder.sendJson(HttpResponseStatus.OK, ConversionHelpers.spec2Summary(
-      instanceService.list(ConversionHelpers.toNamespaceId(namespaceId))));
+      instanceService.list(ConversionHelpers.toNamespaceId(namespaceId), new HashMap<String, String>())));
+  }
+
+  @PUT
+  @Path("/data/datasets/")
```

Comment:
remove trailing slash?

Reply:
its ok to have leading-trailing slashes in the `@Path`. Since for the existing `GET` method we had trailing slash I kept it for this as well.